packages/components/app/styles/components/alert.scss

```diff
@@ -0,0 +1,263 @@
+//
+// ALERT
+//
+// properties within each class are sorted alphabetically
+//
+//
+
+@use "../mixins/focus-ring" as *;
+
+.hds-alert {
+  align-items: flex-start;
+  display: flex;
+}
+
+// ICON
+
+.hds-alert__icon {
+  flex: none;
+  height: 20px;
+  margin-right: 12px;
+  width: 20px;
+
+  .hds-alert--type-compact & {
+    height: 14px;
+    width: 14px;
+  }
+}
+
+// CONTENT (TEXT + ACTIONS + GENERIC)
+
+.hds-alert__content {
+  flex: 1 1 auto;
+}
+
+// TEXT (TITLE & DESCRIPTION)
+
+.hds-alert__text {
+  color: var(--token-color-foreground-warning-on-surface);
+  display: flex;
+  flex-direction: column;
+  font-family: var(--token-typography-body-200-font-family);
+  font-size: var(--token-typography-body-200-font-size);
+  justify-content: center;
+  line-height: var(--token-typography-body-200-line-height);
+
+  .hds-alert--type-compact & {
+    font-family: var(--token-typography-body-100-font-family);
+    font-size: var(--token-typography-body-100-font-size);
+    line-height: var(--token-typography-body-100-line-height);
+  }
+}
+
+.hds-alert__title {
+  font-weight: var(--token-typography-font-weight-semibold);
+}
+
+.hds-alert__description {
+  color: var(--token-color-foreground-primary);
+  font-weight: var(--token-typography-font-weight-regular);
+
+  .hds-alert__title + & {
+    margin-top: 4px;
+  }
+
+  // we add very basic styling for elements that may be injected via the "description" string
+
+  strong {
+    font-weight: var(--token-typography-font-weight-medium);
+  }
+
+  code, pre {
+    font-family: var(--token-typography-font-stack-code);
+  }
+
+  // Notice: in the future this may become a "Link::Inline" component (for now we declare the styles directly here)
+  a {
+    color: var(--token-color-foreground-action);
+    // At the moment the "focus" state is not well defined in design (the one that is in Figma does not work) so we just apply a simple color to the default outline
+    outline-color: var(--token-color-focus-action-external);
+
+    &:hover {
+      color: var(--token-color-foreground-action-hover);
+    }
+
+    &:active {
+      color: var(--token-color-foreground-action-active);
+    }
+  }
+}
+
+// ACTIONS
+
+.hds-alert__actions {
+  align-items: center;
+  display: flex;
+
+  > * {
```

Comment:
[question] are these still necessary now that we've moved to the contextual component approach vs named block? I thought no, but very possible I misunderstood.

Reply:
Yes, we still need this. This is the "solution" :)